crates/types/src/rpc/transactions/broadcasted_invoke_transaction.rs

```diff
@@ -40,17 +55,48 @@ impl BroadcastedInvokeTransaction {
         }
     }
 
-    pub fn create_sir_invoke_function(&self, chain_id: Felt) -> DevnetResult<SirInvokeFunction> {
-        Ok(SirInvokeFunction::new(
-            self.sender_address.into(),
-            EXECUTE_ENTRY_POINT_SELECTOR.clone(),
-            self.common.max_fee.0,
+    pub fn create_blockifier_invoke_transaction(
+        &self,
+        chain_id: Felt,
+    ) -> DevnetResult<InvokeTransaction> {
+        let (entry_point_selector_field, additional_data) = if (QUERY_VERSION_OFFSET
```

Comment:
yes. The difference between fee estimation/simulation and transaction execution is the state modification. So to have the correct transaction hash, the correct txn version should be used.
Also starknet_in_rust made simillar checks for the query verison. With blockifier those checks should be done by us

Reply:
because the other transactions still use method from starknet in rust to compute the transaction hash. Only the invoke transaction doesn't have such method that comes from starknet_in_rust.
`FieldElement::ONE` represent tx version. The query version offset is taken from starknet-rs because its not public and can't be reused

Reply:
After removing completely starknet_in_rust similar ifs will appear to the other transactions

Reply:
We have a constant called `SUPPORTED_TX_VERSION` that could be used in place of `ONE`

Reply:
https://github.com/0xSpaceShard/starknet-devnet-rs/commit/db9c014292fe021fe8459d267b9b25a13876787c